### ISSUE TITLE
Error about custom dataset 

### ISSUE THREAD

**Issue by zml110120** [CityU-AIM-Group/SCAN#3]
Dear author,
It is a fantastic work! I want to use my dataset in your work. How can i do? 
I want to use VOC2012 dataset as source domain, a COCO format of custom dataset as target domain. After modified `paths_catalog.py`, there are some error during evaluation phase. But, my own dataset only have 6 catagory. How can i sovle it?

![image](https://user-images.githubusercontent.com/29172684/206984512-749007b2-9717-4ca5-a38a-6effe7cc30a7.png)

**Comment by wymanCV** [comment]
Hi @zml110120 , thank you so much for your interest in our work. I think that you'd better check two aspects.

1. Check the class number setting in the .yaml config file. This value should be set to C+1 (C is the real class number and should be set to 7 for your case),
https://github.com/CityU-AIM-Group/SCAN/blob/c42d67416fda5c44ee023e14307b63b1ad9890a2/configs/scan/scan_vgg16_cityscapace_to_foggy.yaml#L68

2. Check whether the coco annotation is generated correctly. You can use the official coco [API](https://github.com/cocodataset/cocoapi) to check and make a comparison with the officially provided Cityscapes JSON. I suppose that the category id in ['categories'] seems to be inconsistent with the one referred to in ['annotations'] in your JSON annotation file.

Moreover, to be honest, I think that SCAN is out of date now, while the recent work [SIGMA ](https://github.com/CityU-AIM-Group/SIGMA) is far more efficient and effective. Welcome to using that framework 😀.

**Comment by zml110120** [comment]
It is worked. But i get another error. How can i solve this problem?

![image](https://user-images.githubusercontent.com/29172684/207003314-66af8b74-3b7b-4056-91da-724153037087.png)

**Comment by wymanCV** [comment]
Hi, it seems that you didn't compile the project with GPU correctly. What is your cudatoolkit version and GPU? Do you meet some errors after running "setup.py"?

**Comment by zml110120** [comment]
Hi, i excute `python setup.py build develop` command to compile the project.

![image](https://user-images.githubusercontent.com/29172684/207021310-5fb032e2-931c-468d-9733-30f145e3a969.png)

**Comment by wymanCV** [comment]
@zml110120 Since I notice the "cpu" term appearing in your error, I guess that the project hasn't been compiled with GPU.
![image](https://user-images.githubusercontent.com/93365573/207033698-f756c60a-71b6-4ace-9f60-5683bf30d6b0.png).

Maybe you can check whether your PyTorch can use GPU. Moreover, the compiling process will be very fast in the cpu environment, otherwise, it will take several minutes for the gpu counterpart. You should see lots of "cuda" and "gpu" terms during the compiling instead of "cpu".

**Comment by zml110120** [comment]
@wymanCV I solve this problem. I change `VAL_ITER` in yaml file. But i don't know why it is success. After some iters, I get this error.

![image](https://user-images.githubusercontent.com/29172684/207037046-a893e95c-3488-4275-bdc8-7c779c6c2f33.png)

I only use 5 images to train the model. If i expand the images number, it will be solved. Is it right?

**Comment by wymanCV** [comment]
@zml110120  Hi, maybe it can be solved. 

Actually, there will be many unforeseen problems with custom datasets, for example, the learning rate may not be feasible for some datasets. I have tested the framework on official benchmarks. However, the extension to other custom datasets may not be very easy to adjust. Maybe you need to adjust some hyperparameters, e.g., the learning rate, to train the model. Moreover, it will be good to test the model on the benchmarks provided by us first. Thanks.

**Comment by zml110120** [comment]
Thanks for your patient explain!

**Comment by wymanCV** [comment]
You are welcome and good luck.

**Comment by zml110120** [comment]
@wymanCV Dear author,
If our custom dataset has some catagory which not existing in VOC 2012, whether the SCAN model can get good performance?

**Comment by wymanCV** [comment]
Hi @zml110120 , it should not make any influence. However, this relies on the correctness of your custom dataset establishment. For example, the order of encoded class between training and test must be the same.  Actually, we have tested our method on a custom dataset (see [SCAN++](https://ieeexplore.ieee.org/document/9931144)) and found it can also achieve comparable results. 

For the part, you can directly train a source-only model with pure [FCOS ](https://github.com/tianzhi0549/FCOS) to check whether your dataset is correctly established and whether the detector can use this dataset for object detection. Thanks.

**Comment by wymanCV** [comment]
Thanks for your interest in our work.
Feel free to reopen the issue or contact me through e-mail if you have further problems.
